firehydrant/client.go

```diff
@@ -20,10 +21,27 @@ const (
 	UserAgentPrefix = "firehydrant-terraform-provider"
 )
 
+// Errors
+// NotFound is returned when receiving a 404.
 type NotFound string
 
-func (nf NotFound) Error() string {
-	return string(nf)
+func (err NotFound) Error() string {
+	return string(err)
+}
+
+// checkResponseStatusCode checks to see if the response's status
+// code corresponds to an error or not. An error is returned for
+// all status codes 300 and above
+// TODO: parse response body for other errors
+func checkResponseStatusCode(response *http.Response) error {
+	switch code := response.StatusCode; {
+	case code >= 200 && code <= 299:
+		return nil
+	case code == 404:
+		return NotFound("resource not found")
+	default:
```

Comment:
case code == 401:
  return NotFound("Authorization failure. Check your token")

Reply:
There have been at least 2-3 escalation tickets I've handled that were bad API tokens (deleted, cut/paste issues, not used at all) -- returning a custom message for the 401 on auth failure would really stop that in its tracks